crates/rpc/rpc/src/validation.rs

```diff
@@ -388,14 +391,21 @@ where
             .try_seal_with_senders()
             .map_err(|_| EthApiError::InvalidTransactionSignature)?;
 
-        self.validate_message_against_block(
-            block,
-            request.request.message,
-            request.registered_gas_limit,
-        )
-        .await
-        .map_err(|e| RethError::Other(e.into()))
-        .to_rpc_result()
+        let (tx, rx) = oneshot::channel();
+        let this = self.clone();
+
```

Comment:
what I'd like to here is is wrap the entire function into a spawn_blocking call, effectively duplicating the function: trait + native and then the trait function just calls `Self::`

for example

https://github.com/paradigmxyz/reth/blob/870ffae9094e411343781f7a1dc29d503740f168/crates/rpc/rpc/src/debug.rs#L938-L940

Reply:
Gotcha updated so that we wrap the entire function into a `spawn_blocking` call and calling it using `Self::`